File: array/quick_sort.js
// write a quicksort alg in JS

function quickSort(arr) {
    if (arr.length <= 1) return arr;

    let pivot = [arr[0]]
    let left = arr.slice(1).filter(ele => ele < pivot)
    let right = arr.slice(1).filter(ele => ele >= pivot)

    return quickSort(left).concat(pivot).concat(quickSort(right))
}

console.log(quickSort([5, 3, 7, 2, 1, 8, 0, 10]))

File: array/quick_sort.py
def quick_sort(nums):
    if len(nums) <= 1:
        return nums

    pivot = [nums[0]]
    left = list(filter(lambda x: x < nums[0], nums[1:]))
    right = list(filter(lambda x: x >= nums[0], nums[1:]))

    return quick_sort(left) + pivot + quick_sort(right)

print(quick_sort([6, 2, 5, 1, 4, 8, 7, 3]))

